webapp/packages/core-sdk/src/Resource/CachedDataResource.test.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2023 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { CachedDataResource } from './CachedDataResource';
+
+interface ILoaderData {
+  id: string;
+  value: number;
+}
+
+const LOADER_DATA_MOCK: ILoaderData[] = [
+  {
+    id: '1',
+    value: 1,
+  },
+  {
+    id: '2',
+    value: 2,
+  },
+];
+
+class TestDataResource extends CachedDataResource<ILoaderData[]> {
+  constructor() {
+    super(() => []);
+  }
+
+  private async fetchFromAPI(): Promise<ILoaderData[]> {
+    return new Promise(resolve => {
+      setTimeout(() => {
+        resolve(LOADER_DATA_MOCK);
+      }, 1);
+    });
+  }
+
+  protected async loader() {
+    const data = await this.fetchFromAPI();
+    return data;
+  }
+}
+
+describe('CachedDataResource', () => {
+  let dataResource: TestDataResource;
+
+  beforeEach(() => {
+    dataResource = new TestDataResource();
+  });
+
+  test('should instantiate correctly', () => {
+    expect(dataResource).toBeInstanceOf(CachedDataResource);
+  });
+
+  test('should initialize with an empty array', () => {
+    expect(dataResource.data).toEqual([]);
+  });
```

Comment:
It is better to move it to `const DEFAULT_STATE = () => []` and use this const in the resource constructor and here so the test will be self-described. It's hard to understand why it must be initialized with an empty array when it must be initialized with the default state.